### ISSUE TITLE
[ADD] Keyboard Redragon Indrah

### ISSUE THREAD

**Issue by Sekeye** [antonpup/Aurora#718]
### What you suggest?

- [ ] Aurora
- [ ] Aurora supported game, (Fill in the game)
- [ ] Aurora Wrapper
- [ ] Aurora Profile
- [ ] Aurora supported device, (Fill in problematic devices)
- [ ] Aurora Scripts
- [ ] Aurora Updater
- [ ] Aurora Lighting issue
- [ ] Other, (Fill in another issue here)
- [X] Add support for Redragon Keyboard

### Keyboard:
Redragon Indrah

### Please include any logs, screenshots, or files that are related to this issue:

For screenshots, you can use: http://cdn.pichau.com.br/media/catalog/product/cache/1/thumbnail/440x310/9df78eab33525d08d6e5fb8d27136e95/k/5/k555-7.jpg

**Comment by simon-wh** [comment]
I cannot find any SDK for that keyboard, so there isn't anything I can do to support it.

**Comment by JoshWobbles** [comment]
Reddragon keyboards accept RGB control over the HID protocol, no SDK needed.

**Comment by simon-wh** [comment]
@JoshWobbles Do you have a link to a document or page detailing how to control the lighting of this keyboard over HID?

**Comment by JoshWobbles** [comment]
I have not found one yet, would have to be created using hid analyzers, i took a quick peek and it does look like there is some bi-directional communication going on any time a lighting is changed either on the KB itself via hotkeys or in the software.

**Comment by simon-wh** [comment]
I can't add support for it unless I have a document detailing how to use it or access to the keyboard itself

**Comment by BruhAtTheDesk** [comment]
@simon-wh what would you need to see if this is possible? How can i help?

**Comment by boJob-1** [comment]
Maybe this can help?
[https://gitlab.com/CalcProgrammer1/OpenRGB/-/wikis/Redragon-K556-Devarajas](url)

**Comment by lDeathcrow** [comment]
this can help?
https://github.com/D4rkTT/Redragon-RGB-SDK

**Comment by senposage** [comment]
Interesting so I was under the impression the RedDragon Devices lacked support for any led control this provides some hope that someday we might be-able to replace the software

**Comment by Kizoky** [comment]
My MODECOM keyboard is detected as Redragon K556 Devarajas (via OpenRGB), so if this gets added many MODECOM RGB keyboards might work aswell

**Comment by diogotr7** [comment]
This is not possible due to the firmware of redragon products not being able to receive real time lighting data.

**Comment by CalcProgrammer1** [comment]
We finally got a fairly complete build of QMK running on the Redragon keyboards with the OpenRGB HID protocol working.  Here is a short video of my Redragon K552 running QMK, connected to OpenRGB + Aurora running Overwatch.  The lighting works pretty well.

https://www.youtube.com/watch?v=Pf_Q3B1PmK4

I also tested this on the K556.  Most Redragon keyboards use the same Sonix microcontroller, so they should be supportable.  The stock firmware will never be supported for effects like this, as it lacks a direct control mode and is thus useless for real-time effects.  The only solution is to flash a custom replacement firmware.  We've chosen QMK for this as it already exists, is open source, and already runs on a ton of custom keyboards.
